src/main/java/org/alfresco/rest/api/impl/CommentsImpl.java

```diff
@@ -123,13 +124,26 @@ private Comment toComment(NodeRef nodeRef, NodeRef commentNodeRef, List<String>
         boolean canEdit = map.get(CommentService.CAN_EDIT);
         boolean canDelete =  map.get(CommentService.CAN_DELETE);
 
-		Person createdBy = people.getPerson((String) nodeProps.get(ContentModel.PROP_CREATOR), include);
+		Person createdBy;
+        try {
+			createdBy = people.getPerson((String) nodeProps.get(ContentModel.PROP_CREATOR), include);
+		} catch (EntityNotFoundException enfe){
+        	createdBy = new Person();
+        	createdBy.setUserName((String) nodeProps.get(ContentModel.PROP_CREATOR));
+		}
 		nodeProps.put(Comment.PROP_COMMENT_CREATED_BY, createdBy);
-		
-		Person modifiedBy = people.getPerson((String) nodeProps.get(ContentModel.PROP_MODIFIER), include);
+
+		Person modifiedBy;
+		try {
+			modifiedBy = people.getPerson((String) nodeProps.get(ContentModel.PROP_MODIFIER), include);
+		} catch (EntityNotFoundException enfe)
+		{
+			modifiedBy = new Person();
+			modifiedBy.setUserName((String) nodeProps.get(ContentModel.PROP_MODIFIER));
+		}
 		nodeProps.put(Comment.PROP_COMMENT_MODIFIED_BY, modifiedBy);
-		
-        Comment comment = new Comment(commentNodeRef.getId(), nodeProps, canEdit, canDelete);
+
+		Comment comment = new Comment(commentNodeRef.getId(), nodeProps, canEdit, canDelete);
         return comment;
     }

```

Comment:
Not your code, but a couple of things to tidy up (unused import, double semi-colon) if you feel inclined :)